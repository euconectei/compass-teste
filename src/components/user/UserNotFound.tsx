type UserNotFoundType = {
  username: string;
};

const UserNotFound = ({ username }: UserNotFoundType) => {
  return (
    <div className='alert alert-info'>
      O usuário {username} não foi encontrado.
    </div>
  );
};

export default UserNotFound;
