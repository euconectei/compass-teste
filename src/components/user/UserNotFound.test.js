import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import UserNotFound from './UserNotFound';

const username = 'teste';

describe('Testing UserNotFound Component', () => {
  it('shold render the component', () => {
    const component = renderer.create(<UserNotFound username={username} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('shold render the component with username', () => {
    const { getByText } = render(<UserNotFound username={username} />);
    expect(
      getByText(`O usuário ${username} não foi encontrado.`)
    ).toBeInTheDocument();
  });
});
