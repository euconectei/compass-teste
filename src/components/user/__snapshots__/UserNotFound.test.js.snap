// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing UserNotFound Component shold render the component 1`] = `
<div
  className="alert alert-info"
>
  O usuário 
  teste
   não foi encontrado.
</div>
`;
